### DISCUSSION TITLE
lldb-server

### DISCUSSION THREAD

**Discussion by 0rtz** [mfussenegger/nvim-dap#555] [Q&A] [answered]
Can you please explain whether it is possible to connect to lldb-server using this plugin?
With this configuration:
```lua
local dap = require('dap')
dap.adapters.lldb_server = {
	type = 'server',
	host = '127.0.0.1',
	port = 13000,
}
dap.configurations.cpp = {
	{
		type = 'lldb_server',
		request = 'attach',
		name = 'Connect',
		cwd = '${workspaceFolder}',
	},
}
```

I launch:
`lldb-server g localhost:13000 -- ./program --arg val`

And after issuing `lua require'dap'.continue()` see this lldb-server output:
```
Connection established.
lldb-server exiting...
```

**Comment by mfussenegger** [comment, marked as answer]
lldb-server itself doesn't implement the debug adapter protocol. But a debug adapter like codelldb supports it. See https://github.com/vadimcn/vscode-lldb/blob/54d4fbb2dead83e98bd44d07057b67efc4305531/MANUAL.md#connecting-to-lldb-server-agent

**Comment by 0rtz** [comment]
@mfussenegger Thank you for the answer. I missed wiki about C/C++/Rust (cpptools), workflow I wanted to achieve is starting debuggee with parameters from the shell and connecting with nvim-dap, cpptools gives this ability.
